cmd/goal/application.go

```diff
@@ -195,33 +192,25 @@ func init() {
 	methodAppCmd.Flags().MarkHidden("app-arg") // nolint:errcheck
 }
 
-type appCallBytes struct {
-	Encoding string `codec:"encoding"`
-	Value    string `codec:"value"`
-}
-
-func newAppCallBytes(arg string) appCallBytes {
-	parts := strings.SplitN(arg, ":", 2)
-	if len(parts) != 2 {
-		reportErrorf("all arguments and box names should be of the form 'encoding:value'")
-	}
-	return appCallBytes{
-		Encoding: parts[0],
-		Value:    parts[1],
+func newAppCallBytes(arg string) logic.AppCallBytes {
```

Comment:
@algochoi Double checking:  `newAppCallBytes` is exercised via `cmd/goal/formatting_test.go`.  Since the PR generalizes the app call arg encoding, have you considered moving the tests alongside `data/transactions/logic/parsing.go` (e.g. `parsing_test.go`)?  

If so, I think there's value in retaining a test for `cmd/goal/application.go` proving `newAppCallBytes` errors out on bad input (via `require.Panics`).

Reply:
That's a fair point, I'll move the tests to check the underlying implementation and keep the `Panic` test in `application.go`.

Reply:
@algochoi Thanks for relocating tests via https://github.com/algorand/go-algorand/pull/4183/commits/3ce52f39ba185f62b95a8eba1be32e8843a77fb0 - resolving.